pkg/apis/serving/v1alpha1/revision_lifecycle.go

```diff
@@ -270,3 +271,84 @@ func (r *Revision) GetLastPinned() (time.Time, error) {
 func (rs *RevisionStatus) duck() *duckv1beta1.Status {
 	return &rs.Status
 }
+
+// PropagateDeploymentStatus takes the Deployment status and applies its values
+// to the Revision status.
+func (rs *RevisionStatus) PropagateDeploymentStatus(original *appsv1.DeploymentStatus) {
+	ds := transformDeploymentStatus(original)
+	cond := ds.getCondition(deploymentConditionReady)
+	if cond == nil {
+		return
+	}
+
+	switch cond.Status {
+	case corev1.ConditionUnknown:
+		revCondSet.Manage(rs).MarkUnknown(RevisionConditionResourcesAvailable, cond.Reason, cond.Message)
+	case corev1.ConditionTrue:
+		revCondSet.Manage(rs).MarkTrue(RevisionConditionResourcesAvailable)
+	case corev1.ConditionFalse:
+		revCondSet.Manage(rs).MarkFalse(RevisionConditionResourcesAvailable, cond.Reason, cond.Message)
+	}
+}
+
+var depCondSet = apis.NewLivingConditionSet(
+	deploymentConditionProgressing,
+	deploymentConditionReplicaSetReady,
+)
+
+const (
+	// deploymentConditionReady means the underlying deployment is ready.
+	deploymentConditionReady = apis.ConditionReady
+	// deploymentConditionReplicaSetready inverts the underlying deployment's
+	// ReplicaSetFailure condition.
+	deploymentConditionReplicaSetReady apis.ConditionType = "ReplicaSetReady"
+	// deploymentConditionProgressing reflects the underlying deployment's
+	// Progressing condition.
+	deploymentConditionProgressing apis.ConditionType = "Progressing"
+)
+
+type deploymentStatus struct {
+	duckv1beta1.Status `json:",inline"`
```

Comment:
Why not simply make this an alias?
```
type deploymentStatus duckv1beta1.Status`
```

Why have the `json` bit?

Reply:
I feel like with the `GetCondition` method on `duckv1beta1.Status` and inlining `initializeConditions` that we should be able to eliminate this type and just have your transform function return a vanilla `duckv1beta1.Status`.